### PR TITLE
[SPARK-22049][DOCS] Confusing behavior of from_utc_timestamp and to_utc_timestamp

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Clarify behavior of to_utc_timestamp/from_utc_timestamp with an example

## How was this patch tested?

Doc only change / existing tests

### CODE REVIEW

**Comment by srowen** [comment]
CC @HyukjinKwon as I think we talked about this a long time ago

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -2226,8 +2226,8 @@ setMethod("from_json", signature(x = "Column", schema = "characterOrstructType")
           })
 
 #' @details
-#' \code{from_utc_timestamp}: Given a timestamp, which corresponds to a certain time of day in UTC,
-#' returns another timestamp that corresponds to the same time of day in the given timezone.
+#' \code{from_utc_timestamp}: Given a timestamp like '2017-07-14 02:40:00.0', interprets it as a time in UTC, and
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R]: I think lintr does not recognise this case but I believe these lines should fit in 100 character lengths ..

**Comment by HyukjinKwon** [comment]
LGTM

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -2226,8 +2226,8 @@ setMethod("from_json", signature(x = "Column", schema = "characterOrstructType")
...
+# renders that time as a timestamp in the given time zone. For example, 'GMT+1' would yield '2017-07-14 03:40:00.0'.
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R]: Oh, and missing `#'` at the beginning.

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -2286,8 +2286,8 @@ setMethod("next_day", signature(y = "Column", x = "character"),
           })
 
 #' @details
-#' \code{to_utc_timestamp}: Given a timestamp, which corresponds to a certain time of day
-#' in the given timezone, returns another timestamp that corresponds to the same time of day in UTC.
+#' \code{to_utc_timestamp}: Given a timestamp like '2017-07-14 02:40:00.0', interprets it as a time in the given
+# time zone, and renders that time as a timestamp in UTC. For example, 'GMT+1' would yield '2017-07-14 01:40:00.0'.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: same here with `#` -> `#'`

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**